examples/common/convert_c4.py

```diff
@@ -111,23 +365,39 @@ def main(args: Namespace) -> None:
     Args:
         args (Namespace): Commandline arguments.
     """
-    columns = {'text': 'str', 'timestamp': 'str', 'url': 'str'}
+    splits = ('train', 'train', 'validation')
+    split_names = ('train', 'train_small', 'val')
+
+    mode, max_length, tokenizer, sep_text, columns, split_samples = _get_kwargs(args)
 
-    for (split, split_new_name, expected_num_samples) in [
-        ('train', 'train', 364868892),
-        ('train', 'train_small', 327680),
-        ('validation', 'val', 364608),
-    ]:
+    for (split, split_new_name,
+         expected_num_samples) in zip(splits, split_names, split_samples):
         # Only generate the splits requested
         if split_new_name not in args.splits:
             continue
 
         # Get samples
-        dataset = build_hf_c4_dataset(split=split)
-        samples = generate_samples(dataset=dataset,
-                                   expected_num_samples=expected_num_samples)
+        print('Downloading and formatting dataset...')
+        dataset = build_hf_c4_dataset(split=split,
+                                      mode=mode,
+                                      max_length=max_length,
+                                      sep_text=sep_text,
+                                      tokenizer=tokenizer,
+                                      expected_num_samples=expected_num_samples)
+        samples = dataset.generate_samples()
+
+        # When concatenating text I notice a ~6x slowdown the first time I ran this:
+        #   no concat iterations/s = ~6800 it/s
+        #   with length 10_000 char, that's ~5x longer sequences,
+        #   so we'd expect ~1350 it/s, but observe ~220 it/sec
+        # However, subsequent runs were 5x faster... ¯\_(ツ)_/¯
+        #
+        # Concatenating tokens is, at least on a Mac, much worse
+        #   I got 85 it/s, despite their being half as many sequences
+        #   which is an 80x slowdown
```

Comment:
Maybe worth removing the comments from the file, or cleaning up a bit?

FWIW, I think concatenating text is not ~5x longer sequences, it's probably more like 8-10x longer sequences as the average seq length is quite small. What I would care most about -- on a Linux machine with 64 cores, what's the total time to process `validation` when using `NO_CONCAT` vs. `CONCAT_TEXT`? As long as both are around ~10sec I wouldn't be worried.

Reply:
I will check on a linux box with more cores and remove these comments

Reply:
`CONCAT_TEXT` took 9s on a 64 core machine